test/unit/utils/UWebSocketTransport.spec.ts

```diff
@@ -10,270 +15,693 @@ describe('UWebSocketTransport', () => {
+        'ws_message_size_bytes',
+        { result: 'success' },
+        message.byteLength
+      )
+    })
+
+    it('should record metrics for backpressure events', async () => {
+      mockSocket.send.mockReturnValue(UWebSocketSendResult.BACKPRESSURE)
+      const sendPromise = transport.sendMessage(new Uint8Array([1]))
+      await jest.advanceTimersByTimeAsync(0) // Process initial attempt
 
-      const sendPromises = messages.map((msg) => transport.sendMessage(msg))
+      expect(mockMetrics.increment).toHaveBeenCalledWith('ws_backpressure_events', { result: 'backpressure' })
 
-      // Let the first message process
-      await Promise.resolve()
-      expect(mockSocket.send).toHaveBeenCalledWith(messages[0], true)
+      // Clean up
+      mockSocket.send.mockReturnValue(UWebSocketSendResult.SUCCESS)
+      await jest.runAllTimersAsync()
+      // Cannot safely await/catch if potentially void. Test focuses on metric.
+    })
 
-      // Process second message
-      jest.advanceTimersByTime(1000)
-      await Promise.resolve()
+    it('should record metrics for dropped events', async () => {
+      mockSocket.send.mockReturnValue(UWebSocketSendResult.DROPPED)
+      const sendPromise = transport.sendMessage(new Uint8Array([1]))
+      // Add catch to prevent unhandled rejection
+      ;(sendPromise as unknown as Promise<void>)?.catch(() => {})
 
-      await Promise.all(sendPromises)
-      expect(mockSocket.send).toHaveBeenCalledTimes(2)
+      // Process initial attempt
+      await jest.advanceTimersByTimeAsync(0)
+
+      // Check increment metric
+      expect(mockMetrics.increment).toHaveBeenCalledWith('ws_backpressure_events', { result: 'dropped' })
+
+      // Check size observation metric
+      expect(mockMetrics.observe).toHaveBeenCalledWith('ws_message_size_bytes', { result: 'dropped' }, 1)
+
+      // Clean up
+      await jest.runAllTimersAsync()
     })
 
-    it('should handle queue overflow and drain timeout', async () => {
-      mockSocket.send.mockReturnValue(0) // Force queue to fill
-      const message = new Uint8Array([1])
+    it('should log transport cleanup events correctly', () => {
+      transport.close() // Trigger cleanup
+
+      // Check log message format, including transportId and stringified booleans
+      expect(mockLogs.getLogger('ws-transport').debug).toHaveBeenCalledWith(
+        'Cleaning up transport', // Exact message from code
+        {
+          transportId: expect.any(String),
+          code: 1000,
+          reason: 'Client requested closure',
+          queueLength: 0,
+          isTransportActive: 'true', // Should be true at the time of logging
+          isInitialized: 'true' // Should be true at the time of logging
+        }
+      )
+    })
+  })
 
-      const sendPromise = transport.sendMessage(message)
-      jest.advanceTimersByTime(5000)
+  describe('Transport State Management', () => {
+    it('should handle transport state transitions', async () => {
```

Comment:
Can this happen in the current flow of execution? If the user gets disconnected, doesn't the transport gets re-built?